mgr/src/server/plugins/volume_utils.cr

```diff
@@ -66,21 +66,64 @@ def validate_volume_create(req)
         return NodeResponse.new(false, {"error": "Storage unit parent directory(#{Path[storage_unit.path].parent}) not exists"}.to_json)
       end
 
+      storage_unit_pre_exists = File.exists?(Path[storage_unit.path])
+
+      puts "storage_units_pre_exists: #{storage_unit_pre_exists}"
+      puts "reuse storage_units: #{req.reuse_storage_units}"
+      puts "volume id: #{req.id}"
+
+
+      # Validation to check if creation of storage-unit & xattr are supported or not.
       begin
         Dir.mkdir storage_unit.path
+        xattr = XAttr.new(storage_unit.path)
+        xattr[TEST_XATTR_NAME] = TEST_XATTR_VALUE
       rescue ex : Exception
         return NodeResponse.new(false, {"error": "Failed to create Storage unit path #{storage_unit.path} (Error: #{ex})"}.to_json)
+      rescue ex : IO::Error
+        return NodeResponse.new(false, {"error": "Extended attributes are not supported for #{storage_unit.path} (Error: #{ex})"}.to_json)
+      ensure
+        FileUtils.rmdir storage_unit.path unless storage_unit_pre_exists
       end
 
       begin
+        Dir.mkdir storage_unit.path
+      rescue ex: Exception
         xattr = XAttr.new(storage_unit.path)
-        xattr[TEST_XATTR_NAME] = TEST_XATTR_VALUE
-      rescue ex : IO::Error
-        return NodeResponse.new(false, {"error": "Extended attributes are not supported for #{storage_unit.path} (Error: #{ex})"}.to_json)
-      ensure
-        FileUtils.rmdir storage_unit.path
+        if xattr[VOLUME_ID_XATTR_NAME]
+
+          # if xattr[VOLUME_ID_XATTR_NAME] == req.id
+          #   puts "pass: matching volume id"
+
+          if xattr[VOLUME_ID_XATTR_NAME] != req.id && req.reuse_storage_units == true
```

Comment:
Also note. You cant directly compare the Volume id xattr. since `req.id` will have string format and xattr value in bytes.

```
volume_id = UUID.new(req.id).bytes.to_slice

```

Reply:
changed to string then compared.